Modules/_datetimemodule.c

```diff
@@ -6540,23 +6521,65 @@ PyInit__datetime(void)
     Py_BUILD_ASSERT(DI100Y == 25 * DI4Y - 1);
     assert(DI100Y == days_before_year(100+1));
 
-    us_per_ms = PyLong_FromLong(1000);
-    us_per_second = PyLong_FromLong(1000000);
-    us_per_minute = PyLong_FromLong(60000000);
-    seconds_per_day = PyLong_FromLong(24 * 3600);
+    if (!us_per_ms) {
+        us_per_ms = PyLong_FromLong(1000);
+    }
+    if (!us_per_second) {
+        us_per_second = PyLong_FromLong(1000000);
+    }
+    if (!us_per_minute) {
+        us_per_minute = PyLong_FromLong(60000000);
+    }
+    if (!seconds_per_day) {
+        seconds_per_day = PyLong_FromLong(24 * 3600);
+    }
     if (us_per_ms == NULL || us_per_second == NULL ||
         us_per_minute == NULL || seconds_per_day == NULL)
-        return NULL;
+        return -1;
 
     /* The rest are too big for 32-bit ints, but even
      * us_per_week fits in 40 bits, so doubles should be exact.
      */
-    us_per_hour = PyLong_FromDouble(3600000000.0);
-    us_per_day = PyLong_FromDouble(86400000000.0);
-    us_per_week = PyLong_FromDouble(604800000000.0);
+    if (!us_per_hour) {
+        us_per_hour = PyLong_FromDouble(3600000000.0);
+    }
+    if (!us_per_day) {
+        us_per_day = PyLong_FromDouble(86400000000.0);
+    }
+    if (!us_per_week) {
+        us_per_week = PyLong_FromDouble(604800000000.0);
+    }
     if (us_per_hour == NULL || us_per_day == NULL || us_per_week == NULL)
-        return NULL;
-    return m;
+        return -1;
+
+    // Workaround to allow static PyTypeObjects to have they dict redefined.
+    // XXX: to be removed once _datetime moves to heap allocated PyTypeObjects.
+    PyType_ClearCache();
+
+    return 0;
+}
+
+static PyModuleDef_Slot datetimemodule_slots[] = {
+    {Py_mod_exec, datetimemodule_exec},
+    {0, NULL}
+};
+
+static struct PyModuleDef datetimemodule = {
+    PyModuleDef_HEAD_INIT,
+    "_datetime",
+    "Fast implementation of the datetime type.",
+    0,
```

Comment:
My understanding is that this implies that the module is safe for subinterpreters, but this module still has several process-wide globals. Is it actually appropriate to set `m_size` to `0` in this case?

(Yes, I realize that it cannot be `-1` if you want multi-phase initialization).